### DISCUSSION TITLE
How to format like "const char * fmt, ..."

### DISCUSSION THREAD

**Discussion by gakki2019** [gabime/spdlog#3173] [Q&A]
Hi, 
I want to make a library(linux), and export several C APIs,  one of the API will be:

```
class logger {
public:
     std::shared_ptr<spdlog::logger> m_logger;
 }

static logger theLogger;

extern "C" void mylog_print(int lv, const char* fmt, ...) {
     // how to use spdlog::logger* pointer to handle this ?
     theLogger.m_logger->log(...);
}
```

invoke:  
mylog_print(spdlog::level::debug, "hello {}", "spdlog");
mylog_print(spdlog::level::debug, "hello 1+1 = {}", 2);

**Comment by tt4g** [comment]
It is not possible to export C++ templates as C APIs, because the APIs are not compatible.

EDIT: C++ templates are treated like regular functions if they are specialized, so you can call them from C if you have explicit specialization: [c\+\+ \- How to export template function? \- Stack Overflow](https://stackoverflow.com/questions/68156762/how-to-export-template-function)
But perhaps that is not what you want to do.
